File: GildedRose/src/com/gildedrose/items/SuspiciousItem.java
package com.gildedrose.items;

public class SuspiciousItem extends ItemDecorator {
	
	// Constructors
	protected SuspiciousItem(Item itemToDecorate)
	{
		super(itemToDecorate);
	}

	// Protected methods
	@Override
	protected void updateQuality()
	{
		if (this.isSellInOddNumber())
		{
			if (this.isSellInEnded())
			{
				this.decreaseQualityTwice();
			}
			else
			{
				this.decreaseQualityOnce();
			}		
		}
		else
		{
			if (this.isSellInEnded())
			{
				this.decreaseQualityQuarce();
			}
			else
			{
				this.decreaseQualityTwice();
			}
		}
	}

}

File: GildedRose.Net/Items/SuspiciousItem.cs
namespace GildedRose.Net.Items
{
    public class SuspiciousItem : ItemDecorator
    {
    
        internal SuspiciousItem(Item itemToDecorate) : base(itemToDecorate) { }
        
        // Protected methods
        protected override void UpdateQuality()
        {
            if (this.SellInOddNumber)
            {
                if (this.SellInEnded)
                {
                    this.DecreaseQualityTwice();
                }
                else
                {
                    this.DecreaseQualityOnce();
                }        
            }
            else
            {
                if (this.SellInEnded)
                {
                    this.DecreaseQualityQuarce();
                }
                else
                {
                    this.DecreaseQualityTwice();
                }
            }
        }
        
    }
}